### ISSUE TITLE
[Feat]: Add OpenTelemetry instrumentation for monitoring pgvector

### ISSUE THREAD

**Issue by amanagarwal042** [openlit/openlit#282]
### 🚀 What's the Problem?

Cannot monitor pgvector based Applications

### 💡 Your Dream Solution

Similar to Pinecone, Have a auto instrumentation for pgvector

### 🤔 Seen anything similar?

NA

### 🖼️ Pictures or Drawings

NA

### 👐 Want to Help Make It Happen?
- [ ] Yes, I'd like to volunteer and help out with this!

**Comment by k4kum4r** [comment]
Hi @patcher9 ,
I'm new to OSS and would love to make a contribution. Can I take this up?

**Comment by amanagarwal042** [comment]
For sure you can pick this up!

I have assigned you the issue, Incase you wanna brainstorm/discuss/ask questions about the implementation, Feel free to ask here or on our [community slack](https://join.slack.com/t/openlit/shared_invite/zt-2etnfttwg-TjP_7BZXfYg84oAukY8QRQ),

**Comment by k4kum4r** [comment]
Is there a reference implementation of setting up instrumentation for other databases?
From what I can see [this](https://github.com/openlit/openlit/blob/main/sdk/python/src/openlit/instrumentation/pinecone/pinecone.py) seems to be for Pinecone, correct?

**Comment by amanagarwal042** [comment]
Yup, Pinecone is a good example, you can also check out [ChromaDB Instrumentation](https://github.com/openlit/openlit/tree/main/sdk/python/src/openlit/instrumentation/chroma), [Milvus Instrumentation](https://github.com/openlit/openlit/tree/main/sdk/python/src/openlit/instrumentation/milvus), [Qdrant Instrumentation](https://github.com/openlit/openlit/tree/main/sdk/python/src/openlit/instrumentation/qdrant). Happy to help as you build this instrumentation! Don't hestiate to ask

**Comment by amanagarwal042** [comment]
Hey @k4kum4r, following up to know if you have an update/blockers/anything I can help with

**Comment by k4kum4r** [comment]
Hi @patcher9 , apologies for the delay.
Dealing with a ton of work. Don't want to hold up this issue. You can unassign me for the time being. I'll come around and look at this if it still remains open.
Thanks!

**Comment by amanagarwal042** [comment]
Sounds good, No Worries!
